data/tabular/train_test_split.py

```diff
@@ -1,209 +1,459 @@
-                print(message)
-            except KeyError:
-                print(f"No split column in {path}")
-
-            merged_data = pd.merge(read_dataset, train_test_df, on=repr_col, how="left")
-            # merged_data = merged_data.dropna()
-            if override:
-                merged_data.to_csv(path, index=False)
-            else:
-                # rename the old data_clean.csv file to data_clean_old.csv
-                os.rename(path, path.replace(".csv", "_old.csv"))
-                # write the new data_clean.csv file
-                merged_data.to_csv(path, index=False)
 
-            if len(merged_data.query("split == 'train'")) == 0:
-                raise ValueError("Split failed, no train data")
-            if len(merged_data.query("split == 'test'")) == 0:
-                raise ValueError("Split failed, no test data")
-            if check:
-                test_split_smiles = set(
-                    merged_data.query("split == 'test'")[repr_col].to_list()
-                )
-                if len(train_smiles.intersection(test_split_smiles)) > 0:
-                    raise ValueError("Split failed, train and test overlap")
+def get_yaml_files(data_dir: Union[str, Path]) -> List[str]:
+    """Returns all yaml files in the data_dir directory."""
+    return glob(os.path.join(data_dir, "**", "*.yaml"), recursive=True)
 
 
-TRACKED_DATASETS = []
+def run_transform(file):
+    if not os.path.exists(os.path.join(os.path.dirname(file), "data_clean.csv")):
+        subprocess.run(
+            ["python", "transform.py"],
+            cwd=os.path.dirname(file),
+        )
 
 
-def per_repr(
-    repr_col: str,
+def remaining_split(
+    data_dir,
+    override: bool = False,
     seed: int = 42,
-    train_size: float = 0.8,
-    val_size: float = 0.0,
-    test_size: float = 0.2,
-    path: str = "*/data_clean.csv",
+    debug: bool = False,
+    train_frac: float = 0.7,
+    val_frac: float = 0.15,
+    test_frac: float = 0.15,
+):
+    # make deterministic
+    np.random.seed(seed)
+    random.seed(seed)
+    dask_random_state = da.random.RandomState(seed)
+
+    yaml_files = get_yaml_files(data_dir)
+    non_smiles_yaml_files = [
+        file for file in yaml_files if not has_yaml_file_smiles_column(file)
+    ]
+
+    # if we debug, we only run split on the first 5 datasets
+    if debug:
+        non_smiles_yaml_files = non_smiles_yaml_files[:5]
+
+    def assign_split(train_frac, val_frac, test_frac, random_state):
+        # Assign random number for train/remaining split using the random state
+        random_value = random_state.random_sample()
+        if random_value < train_frac:
+            return "train"
+        elif random_value < train_frac + test_frac:
+            return "test"
+        else:
+            return "valid"
+
+    # we run random splitting for each dataset
+    for file in non_smiles_yaml_files:
+        print(f"Processing {file}")
+        run_transform(file)
```

Comment:
better check here if `meta.yaml` and `data_clean.csv` exist, see my comment above

Reply:
You mean instead of `run_transform`?